### PR TITLE
WP: Stacklight integrations

### PR DESCRIPTION
**Pull request by cznewt**

### CODE REVIEW

File: rabbitmq/meta/heka.yml [hunk 1]
```diff
@@ -1,15 +1,53 @@
...
+      differentiator: ["rabbitmq.", "Node"]
+      decoder: "rabbitmq_decoder"
+      splitter: "rabbitmq_splitter"
+  splitter:
+    rabbitmq_splitter:
+      engine: token
+      delimiter: "\n\n(=[^=]+====)"
+      delimiter_eol: false
+metric_collector:
+  filter:
```
Comment (simonpasquier) [inline rabbitmq/meta/heka.yml:23]: This filter and the one below will be generated from the alarms that will be defined in the user model. At least this is our current thinking. Obviously we need to flesh out the details and @elemoine is working on it.
So I would drop the metric_collector part for now and focus on the collectd plugins.

Comment (cznewt) [reply rabbitmq/meta/heka.yml:23]: Alarm definition should be part of service level metadata models, user may be able to override the parameters but they are autogenerated at default.

Comment (elemoine) [reply rabbitmq/meta/heka.yml:23]: @cznewt exactly our thinking.

Comment (elemoine) [reply rabbitmq/meta/heka.yml:23]: @cznewt when you say "service level metadata models" do you mean here in `rabbitmq/meta/heka.yml` or in some class file in `metadata/service/`? On our side we were thinking about the latter (`metadata/service/`), which is what the user can override as he likes.

Comment (SwannCroiset) [reply rabbitmq/meta/heka.yml:23]: We have a requirement as mentioned by @elemoine: the operator must be able to add/disable/override alarm definitions in the salt-model repo. 

@cznewt I don't think having alarm definitions in <service>/meta/*yml allows the operator to override alarm definitions ?

The heka formulas must consume pillar data (reclass parameters) to generate heka sandbox filters (so called AFD filters + alarm definitions in LUA) for services.
We must follow in someway the actual heka formulas shipping its own heka plugins [0] [1] but declared/classified by other services.
[0] https://github.com/tcpcloud/salt-formula-heka/blob/master/heka/server.sls#L184-L199
[1] https://github.com/tcpcloud/salt-formula-heka/blob/master/metadata/service/server/elastic/shipper.yml

Comment (cznewt) [reply rabbitmq/meta/heka.yml:23]: @elemoine @SwannCroiset The support metadata reside inside the salt formulas and get executed within the salt runtime with variables present as are when configuring the service and have access to any system call's or salt module run's outputs. This gives you limitless power to generate whatever you need to you config. Having the default alarms definitions in the same format as other filters removes complexity.
Yes all of the defined models will be  overridable from user defined metadata, the power resides that you have well engineered checks and logic out of box and you can tune it up per need.

Comment (simonpasquier) [reply rabbitmq/meta/heka.yml:23]: @cznewt I think we're aligned. We "just" need to develop further our vision on how the alarms artefacts will be generated... As stated initially @elemoine is preparing sthg on the topic.

File: rabbitmq/meta/collectd.yml [hunk 2]
```diff
@@ -1,14 +1,12 @@
+{%- from "rabbitmq/map.jinja" import server with context %}
 plugin:
- # rabbitmq_server_rabbitmq:
-  #  plugin: rabbitmq
-   # interval: 60
-    #template: rabbitmq/files/collectd_rabbitmq.conf
+  rabbitmq_info:
+    plugin: python
+    execution: local
+    template: collectd/files/collectd_check_local_endpoint.conf
```
Comment (simonpasquier) [inline rabbitmq/meta/collectd.yml]: this should be "rabbitmq/files/collectd_rabbitmq_info.conf" IIUC

File: rabbitmq/meta/collectd.yml [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
+    username: {{ server.admin.username }}
+    password: {{ server.admin.password }}
```
Comment (simonpasquier) [inline rabbitmq/meta/collectd.yml:8]: In general, shall we pass every parameter used by the template like you've done it here for username & password?

**Review by simonpasquier — CHANGES_REQUESTED** [review]

File: rabbitmq/files/collectd_rabbitmq_info.conf [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{%- from "rabbitmq/map.jinja" import server with context %}
```
Comment (simonpasquier) [inline rabbitmq/files/collectd_rabbitmq_info.conf:1]: Since `server` isn't used this line can be removed

**Pull request closed without merging**